platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/LoggerImpl.java

```diff
@@ -81,21 +74,12 @@ public class LoggerImpl implements Logger {
     protected LoggerImpl(
             @NonNull final String name,
             @Nullable final Marker marker,
-            @NonNull final Map<String, String> context,
+            @Nullable final Map<String, String> context,
             @NonNull final LogEventFactory logEventFactory,
             @NonNull final LogEventConsumer logEventConsumer) {
-        if (name == null) {
-            EMERGENCY_LOGGER.logNPE("name");
-            this.name = "";
-        } else {
-            this.name = name;
-        }
+        this.name = Objects.requireNonNull(name, "name must not be null");
```

Comment:
Please do not do that. This would end in throwing an exception at logger creation. The logging API should never throw an exception when used.

Reply:
Perfect one discussion topic I was interested in having:
It makes sense, but what is the difference with the other parameters on the method, e.g., `logEventFactory` and `logEventConsumer`? Can't we assume that if those will not be null (because they need to be handled beforehand and sent a valid parameter before it reaches this method), the same will happen to "name"?